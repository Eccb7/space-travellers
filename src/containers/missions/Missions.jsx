import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { MissionFeature } from '../../components';
import './missions.css';
import { fetchMissions } from '../../redux/missions/missionSlice';

function Missions() {
  const { loading, missions, error } = useSelector((state) => state.mission);

  const dispatch = useDispatch();
  useEffect(() => {
    if (missions.length === 0) {
      dispatch(fetchMissions());
    }
  }, [dispatch, missions.length]);

  if (loading) return (<p>Please wait...</p>);
  if (error) return (<p>{error}</p>);
  return (
    <table className="app_mission-table">
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th>{' '}</th>
        </tr>
      </thead>
      <tbody>
        {
          missions.map((mission) => (
            <MissionFeature
              key={mission.mission_id}
              missionId={mission.mission_id}
              missionName={mission.mission_name}
              missionDesc={mission.description}
              missionReserved={mission.reserved}
            />
          ))
        }
      </tbody>
    </table>
  );
}

export default Missions;
