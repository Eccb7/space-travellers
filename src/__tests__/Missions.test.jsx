import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import configureStore from 'redux-mock-store';
import Missions from '../containers/missions/Missions';

const mockStore = configureStore();

test('matches snapshot', () => {
  const initialState = {
    mission: {
      loading: false,
      missions: [
        {
          mission_id: 'mission1',
          mission_name: 'Mission 1',
          description: 'Mission 1 description',
          reserved: false,
        },
        {
          mission_id: 'mission2',
          mission_name: 'Mission 2',
          description: 'Mission 2 description',
          reserved: true,
        },
      ],
      error: '',
    },
  };

  const store = mockStore(initialState);

  const { container, getByText } = render(
    <Provider store={store}>
      <Missions />
    </Provider>,
  );

  expect(container).toMatchSnapshot();
  expect(getByText('Mission 1')).toBeInTheDocument();
  expect(getByText('Mission 1 description')).toBeInTheDocument();
  expect(getByText('Mission 2')).toBeInTheDocument();
  expect(getByText('Mission 2 description')).toBeInTheDocument();
});
