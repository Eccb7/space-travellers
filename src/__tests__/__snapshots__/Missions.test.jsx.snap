// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <table
    class="app_mission-table"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="app_mission-title"
        >
          Mission 1
        </td>
        <td>
          Mission 1 description
        </td>
        <td>
          <span
            class="app_mission-status app_mission-status_nomember"
          >
            NOT A MEMBER
          </span>
        </td>
        <td>
          <button
            class="app_mission-button app_mission-button_join"
            title="Join Mission"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
      <tr>
        <td
          class="app_mission-title"
        >
          Mission 2
        </td>
        <td>
          Mission 2 description
        </td>
        <td>
          <span
            class="app_mission-status app_mission-status_member"
          >
            Active Member
          </span>
        </td>
        <td>
          <button
            class="app_mission-button app_mission-button_leave"
            title="Leave Mission"
            type="button"
          >
            Leave Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`should render missions correctly 1`] = `
<div>
  <table
    class="app_mission-table"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
        <th>
           
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          class="app_mission-title"
        >
          Mission 1
        </td>
        <td>
          Mission 1 description
        </td>
        <td>
          <span
            class="app_mission-status app_mission-status_nomember"
          >
            NOT A MEMBER
          </span>
        </td>
        <td>
          <button
            class="app_mission-button app_mission-button_join"
            title="Join Mission"
            type="button"
          >
            Join Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
